Sources/OpenAPIRuntime/Errors/RuntimeError.swift

```diff
@@ -96,6 +100,20 @@ internal enum RuntimeError: Error, CustomStringConvertible, LocalizedError, Pret
             return "Transport failed with error: \(underlyingError.localizedDescription)"
         case .handlerFailed(let underlyingError):
             return "User handler failed with error: \(underlyingError.localizedDescription)"
+        case .unexpectedResponseStatus(let expectedStatus, let response):
+            return "Unexpected response, expected status code: \(expectedStatus), response: \(response)"
+        case .unexpectedResponseBody(let expectedContentType, let response):
+            return "Unexpected response body, expected content type: \(expectedContentType), response: \(response)"
         }
     }
 }
+
+@_spi(Generated)
+public func throwUnexpectedResponseStatus(expectedStatus: String, response: Any) throws -> Never {
+    throw RuntimeError.unexpectedResponseStatus(expectedStatus: expectedStatus, response: response)
+}
+
+@_spi(Generated)
+public func throwUnexpectedResponseBody(expectedContent: String, body: Any) throws -> Never {
+    throw RuntimeError.unexpectedResponseBody(expectedContent: expectedContent, body: body)
+}
```

Comment:
I think these could go into `ErrorExtensions.swift`, and maybe return the constructed error rather than throw it (similar to `DecodingError.failedToDecodeAnySchema(...)` etc).

Reply:
Happy to explore a better home for these, but is `ErrorExtensions.swift` a good fit?

In `ErrorExtensions.swift` we extend the _public_ type `DecodingError`. Here we are creating an _internal_ `RuntimeError`. Had this been a public type, I'd have made these SPIs extensions of `RuntimeError`.

Reply:
True, maybe the whole file should be renamed to ErrorUtilities.swift or something, or a new ErrorFactories.swift file could hold these. Will leave up to you.